### DISCUSSION TITLE
twinBasic Announcement at Hacker News?

### DISCUSSION THREAD

**Discussion by jpbro** [twinbasic/twinbasic#837] [General]
About a week ago, I was pleasantly surprised to see some lively discussion [at Hacker News about RAD Basic:](https://news.ycombinator.com/item?id=31282195)

I'm a twinBASIC backer of course, but the comments were very interesting to read and I think they will be of value to @WaynePhillipsEA  (if he hasn't read them yet). 

Do you think it would be worthwhile to submit twinBASIC to Hacker News? It might give the project some extra exposure, but I don't know if that's the kind of exposure that tB wants yet?

**Comment by VB6-Programming** [comment]
I'd suggest waiting until we have the ability to import an existing VB6 application (with forms and ActiveXs) and have that run without modification.
There are lots of 'almost' languages (several mentioned in that discussion). It will be far more impressive to show the ability to import a VB6 program and run it in a modern "VB" while retaining the RAD/GUI capability that many seem to miss in newer languages.

**Reply by jpbro** [reply]
That's a good point!

**Reply by wqweto** [reply]
That's not coming soon though IMO.

From README:
```
With twinBASIC we aim for 100% backwards compatibility with existing VB6 and VBA projects.
```

How about to devise a back-compat metric as a percentage running up to 100%? Are we above 90% yet or not?

**Reply by VB6-Programming** [reply]
> That's not coming soon though IMO.
> 
OK, if not today then tomorrow will be fine ;)

**Reply by DinyaZ** [reply]
"Welcome!
twinBASIC is a modern version of the classic BASIC programming language.  With twinBASIC we aim for 100% backwards compatibility with existing VB6 and VBA projects.
(...)
**Our top priority has been to make sure that we copy every last quirk (or feature?) found in the legacy VB6 compiler.**"
.
If somebody opens the twinbasic.com, can read directly this statement. 
So I hope, the 100.0% VB6 compatibility is the main goal now, and any new functionality comes only after this. If we speak about the VB6 compatibility with the form designer, etc, we are maybe on max 60-70%, but I am very positive, because Wayne does an excellent job and the Twinbasic evolves rapidly.

**Reply by saturday220514** [reply]
I think Wayne wrote that it's more efficient to add new stuff at the same time as working on implementing VB6/VBA features.
Who knows how long it will take but the speed he's moving this along is extraordinary in my view.
In my humble opinion it wouldn't do any harm to give it a bit of publicity on Hacker News at this stage as well as when everything's done.

**Comment by WaynePhillipsEA** [comment]
Someone just posted it on Hacker News today:  https://news.ycombinator.com/item?id=31976614

**Comment by mansellan** [comment]
When the official announcement goes out, it might also be worth posting to Reddit and The Register. And maybe others? Back in the day I would have said also Slashdot, but that site has gone to ruin IMO.

The Register has mentioned twinBASIC a couple of times, but only in passing as links in a couple of update stories to RAD Basic. Didn't look like any commenters picked up on it.

**Reply by VB6-Programming** [reply]
i-programmer.info posted a number of VB6 favourable articles.

**Reply by mansellan** [reply]
I think the official announcement should happen when v1 lands. Doubtless there will be some who notice (like HN did) in the meantime, but v1 is the epoch.

Up to @WaynePhillipsEA how he wants to do comms of course,  just my thoughts.

**Reply by mansellan** [reply]
Just commented on the HN post, hope it's ok!

**Reply by WaynePhillipsEA** [reply]
Thanks, both of you for the comments on the HN post.  I was contemplating commenting myself, but I think I'll wait until the v1 release.
